tests/test_ensemble_tools.py

```diff
@@ -0,0 +1,163 @@
+"""Tests for the ensemble_tools utilities."""
+
+import itertools
+import re
+import shutil
+from pathlib import Path
+
+import pytest
+from defusedxml import ElementTree
+from py.path import local
+
+# from plugins.FabNESO.utils.make_sweep_dir
+from FabNESO.ensemble_tools import create_dict_sweep, create_dir_tree, edit_parameters
+
+
+def test_edit_parameters(tmpdir: local) -> None:
+    """Test the edit_parameters method of ensemble_tools."""
+    parameter_to_test = "particle_initial_velocity"
+    parameter_test_value = 10.0
+    temp_conditions_path = Path(tmpdir) / "conditions.xml"
+    shutil.copyfile(
+        Path("config_files") / "two_stream" / "conditions.xml", temp_conditions_path
+    )
+
+    edit_parameters(temp_conditions_path, {parameter_to_test: parameter_test_value})
+
+    assert _check_parameter_in_conditions(
+        temp_conditions_path, parameter_to_test, parameter_test_value
+    )
+
+
+def _check_parameter_in_conditions(
+    conditions_file_name: Path, parameter_name: str, expected_value: float
+) -> bool:
+    """Return True if parameter_name has approx expected_value."""
+    data = ElementTree.parse(conditions_file_name)
+    root = data.getroot()
+    conditions = root.find("CONDITIONS")
+    if conditions is None:
+        msg = f"Failed to find CONDITIONS in the file {conditions_file_name}"
+        raise ValueError(msg)
+    parameters = conditions.find("PARAMETERS")
+    if parameters is None:
+        msg = f"Failed to find PARAMETERS in the \
+        CONDITIONS node of {conditions_file_name}"
+        raise ValueError(msg)
+
+    for element in parameters.iter("P"):
+        match = re.match(
+            r"\s*(?P<key>\w+)\s*=\s*(?P<value>-?\d*(\.?\d)+)\s*", element.text
+        )
+        if match is None:
+            msg = f"Parameter definition of unexpected format: {element.text}"
+            raise ValueError(msg)
+        key = match.group("key")
+        value = float(match.group("value"))
+        if key == parameter_name:
+            return expected_value == pytest.approx(value)  # Because of rounding errors
+
+    # If we make it this far then we haven't found the parameter we were looking for,
+    # which is surely a problem in its own right
+    msg = f"Could not find the parameter {parameter_name} in {conditions_file_name}"
+    raise ValueError(msg)
```

Comment:
I would say better to return `False` in the case that parameter not found at all in file, and then have the `assert` statements in test handle this as a test failure. Generally tests are expected to fail by asserting something false rather than raise exceptions (which will be shown as a test error rather than exception).

Reply:
See previous discussion. This would then return `(0,0)`, which can be a failed `assert` in a subsequent test.